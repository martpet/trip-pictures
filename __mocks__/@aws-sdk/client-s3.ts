export const S3Client = function DummyS3Client() {};
