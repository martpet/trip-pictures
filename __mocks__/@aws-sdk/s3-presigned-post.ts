export const createPresignedPost = vi
  .fn()
  .mockName('createPresignedPost')
  .mockResolvedValue('dummyPresignedPost');
