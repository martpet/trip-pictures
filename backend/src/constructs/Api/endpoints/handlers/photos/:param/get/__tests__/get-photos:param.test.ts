import { DynamoDBDocumentClient, GetCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import {
  itResolves,
  itResolvesWithError,
  itSendsAwsCommand,
} from '~/constructs/Api/utils';

import { handler } from '../get-photos:param';

vi.mock('~/constructs/Api/utils/errorResponse');

const ddbMock = mockClient(DynamoDBDocumentClient);

beforeEach(() => {
  ddbMock.reset();
  ddbMock.on(GetCommand).resolves({ Item: { dummyPhotoProp: 'dummyPhotoValue' } });
});

const args = [
  { pathParameters: { fingerprint: 'dummyFingerprint' } },
] as unknown as Parameters<typeof handler>;

describe('get-photos:param', () => {
  itSendsAwsCommand(GetCommand, ddbMock, handler, args);
  itResolves(handler, args);

  describe('when "Item" is missing from "GetCommand" output', () => {
    beforeEach(() => {
      ddbMock.on(GetCommand).resolves({});
    });
    itResolvesWithError(handler, args);
  });
});
