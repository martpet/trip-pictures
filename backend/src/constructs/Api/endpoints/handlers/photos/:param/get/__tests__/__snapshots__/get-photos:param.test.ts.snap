// Vitest Snapshot v1

exports[`get-photos:param > resolves with a correct value 1`] = `
{
  "dummyPhotoProp": "dummyPhotoValue",
}
`;

exports[`get-photos:param > sends "GetCommand" from "DynamoDBDocumentClient" with correct args 1`] = `
{
  "Key": {
    "fingerprint": "dummyFingerprint",
  },
  "TableName": "TP-photos",
}
`;

exports[`get-photos:param > when "Item" is missing from "GetCommand" output > calls "errorResponse" with correct args 1`] = `
[
  [
    "565cca53ca",
    {
      "statusCode": 404,
    },
  ],
]
`;

exports[`get-photos:param > when "Item" is missing from "GetCommand" output > resolves with a correct value 1`] = `"dummyErrorResponse"`;
